### DISCUSSION TITLE
Is it safe to use both SignedCookieJar and CookieJar in a single handler?

### DISCUSSION THREAD

**Discussion by yuttie** [tokio-rs/axum#2036] [Q&A] [answered]
I'm writing a handler like the one below and it actually works as intended.
But I'm not sure it's safe or not.

Please note that `sjar` and `jar` never share the same key, e.g. "ABC" and "XYZ" as shown in the code.

```rust
pub async fn handler(
    sjar: SignedCookieJar,
    jar: CookieJar,
) -> Result<(SignedCookieJar, SignedCookieJar), MyAppError> {
    ...
    let sjar = sjar.add(Cookie::new("ABC", ...));
    let jar = jar.add(Cookie::new("XYZ", ...));
    ...
    Ok((sjar, jar))
}
```

**Comment by davidpdrsn** [comment, marked as answer]
Yes that is safe, even if they share the same key.

Cookies work via the `Set-Cookie` header which contains a key/value pair like `Set-Cookie: key1=value1;`. So if you set `ABC` in a signed jar that translates to a signed value in the header. If you then set `XYZ` in a different header those don't conflict.

**Reply by yuttie** [reply]
@davidpdrsn Thank you for your answer. :smiley: 

Your explanation helped me understand how the cookie jars are translated into headers, so I understand that there is no conflict, at least for my use with different keys.
